src/peft/tuners/boft/config.py

```diff
@@ -0,0 +1,124 @@
+            and saved in the final checkpoint.
+        layers_to_transform (`Union[List[int],int]`):
+            The layer indexes to transform, if this argument is specified, it will apply the BOFT transformations on
+            the layer indexes that are specified in this list. If a single integer is passed, it will apply the BOFT
+            transformations on the layer at this index.
+        layers_pattern (`str`):
+            The layer pattern name, used only if `layers_to_transform` is different from `None` and if the layer
+            pattern is not in the common layers pattern.
+    """
+
+    boft_block_size: int = field(
+        default=0,
+        metadata={
+            "help": "BOFT block size across different layers.",
+            "note": "You can only specify either boft_block_size or boft_block_num, but not both simultaneously, because boft_block_size x boft_block_num = layer dimension.",
+        },
+    )
+    boft_block_num: int = field(
+        default=0,
+        metadata={
+            "help": "Number of BOFT blocks per injected layer.",
+            "note": "You can only specify either boft_block_size or boft_block_num, but not both simultaneously, because boft_block_size x boft_block_num = layer dimension.",
+        },
+    )
+    boft_n_butterfly_factor: int = field(
+        default=1,
+        metadata={
+            "help": "Number of butterfly factors.",
+            "note": (
+                "for example, boft_n_butterfly_factor=2, the effective block size of OFT becomes twice as big and the number of blocks become half.",
+                "note: for boft_n_butterfly_factor=1, BOFT is the same as vanilla OFT.",
+            ),
+        },
+    )
+    target_modules: Optional[Union[List[str], str]] = field(
+        default=None,
+        metadata={
+            "help": "List of module names or regex expression of the module names to replace with BOFT.",
+            "example": "For example, ['q', 'v'] or '.*decoder.*(SelfAttention|EncDecAttention).*(q|v)$' ",
+        },
+    )
+    boft_dropout: float = field(default=0.0, metadata={"help": "BOFT multiplicative dropout"})
+    fan_in_fan_out: bool = field(
+        default=False,
+        metadata={"help": "Set this to True if the layer to replace stores weight like (fan_in, fan_out)"},
+    )
+    bias: str = field(default="none", metadata={"help": "Bias type for BOFT. Can be 'none', 'all' or 'boft_only'"})
+    modules_to_save: Optional[List[str]] = field(
+        default=None,
+        metadata={
+            "help": "List of modules apart from BOFT layers to be set as trainable and saved in the final checkpoint. ",
+            "note": (
+                "For example, in Sequence Classification or Token Classification tasks, ",
+                "the final layer `classifier/score` are randomly initialized and as such need to be trainable and saved.",
+            ),
+        },
+    )
+    init_boft_weights: bool = field(
```

Comment:
Let's rename this argument `init_weights`. I know we're not quite consistent in PEFT, sometimes we have `init_weights` and sometimes `init_<name>_weights`. Using the former is better though, because we can use the same argument name everywhere. Case in point, some unit tests are failing because of:

> TypeError: `BOFTConfig.__init__()` got an unexpected keyword argument 'init_weights'

Reply:
Thanks for the comment. Fixed.